### DISCUSSION TITLE
Tesla Announces API Pricing: Third-Party Service Costs Expected to Rise

### DISCUSSION THREAD

**Discussion by 10tribu** [teslamate-org/teslamate#4426] [General]
<p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">After several years of allowing third-party developers and tinkerers free access to the Tesla API, Tesla has finally announced a pricing for their API. Back in 2023, Tesla introduced their<span> </span><a href="https://www.notateslaapp.com/news/1653/tesla-creates-official-apis-for-third-party-services-to-start-charging-for-usage" style="box-sizing: border-box; color: var(--link-active-color); text-decoration: none; font-weight: 600; border-bottom: 1px solid var(--link-active-color-border);">new API system</a>, along with them being “temporarily free.” </p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Back then, Tesla announced their Discovery tier, which had some limitations but set the groundwork for what was to come. For the first time, developers had official documentation and support, but pricing still was to be determined.</p><div style="box-sizing: border-box; margin-left: 81.2188px; margin-right: 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></div><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tesla recently announced pricing for its APIs, which, unfortunately, is much higher than many were expecting. We’ll start by reviewing everything that’s been announced and then explore what could happen for both home users and larger third-party apps.</p><h2 style="box-sizing: border-box; line-height: 1.2; font-size: 28px; margin: 20px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Fleet API - Usage Based Pricing</h2><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tesla is implementing a pay-per-use pricing model that charges users based on overall usage. Some items, such as Streaming Signals and Commands, are at a lower cost, which helps to incentivize developers to be smart with the data they’re pulling from vehicles. Meanwhile, vehicle data via REST APIs and waking up vehicles is much more expensive.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Here’s the pricing chart:</p><div class="inline" style="box-sizing: border-box; overflow-x: auto; margin: 0px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">
Data Type | Signals/Requests | Cost
-- | -- | --
Streaming Signals | 150,000 | $1
Commands | 1,000 | $1
Data | 500 | $1
Wakes | 50 | $1

</div><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tesla seems like they want developers to focus on what’s happening in the now - rather than being able to track a long driving session or pull charging history from the vehicle. While some things seem cheap and some seem expensive, the overall cost is still fairly high, even for limited personal use.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Additionally, the new Fleet Telemetry doesn’t offer all the data points that are available from the older vehicle data API. That means that you won’t always be able to take advantage of the cheaper “Streaming Signals” data type - but have to fall back to the data package - which is far more expensive.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Many services use REST APIs for vehicle data at about a one-minute interval, which means that each hour of driving or charging could cost about $0.12. Since the average vehicle drives for about an hour a day and charges for an hour or two, that could cost about $0.36 cents per day — per vehicle.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Looking at a month’s worth, that’s about $10/month per vehicle. While the streaming signal API is much cheaper, the total cost would vary depending on the interval and data being collected. Each signal is a piece of data, so if you’re tracking the vehicle’s speed, location, odometer, state and battery level, that’s five signals.</p><h2 style="box-sizing: border-box; line-height: 1.2; font-size: 28px; margin: 20px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Fleet API - Personal Use</h2><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Tesla’s Fleet API provides each Tesla account with a $10 monthly discount - which Tesla says can cover data streaming, 100 commands, and two wakes per day for two vehicles per month. There isn’t a discount per vehicle - it's per account, so if you have more vehicles, it’s less beneficial to you. This should allow home tinkerers to create some automations and track some data without having to pay for access.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">However, this is a fairly restrictive amount of signals/requests for personal use - let’s look at the author’s Home Assistant integration for a fairly simple example.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">On average, with a Home Assistant tablet at home to start/stop charging, check charge states, precondition vehicles, send Google Map targets, and a few more simple commands for just two vehicles - we make, on average, about 200 requests a week per vehicle. That comes out to 1,600 requests a month and 400 wake-ups a month. </p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">We’re also requesting data from the Data portion of the API - not the Streaming Signals portion of the API - and that’s about every five minutes while charging - so a wake isn’t needed. With 480 data requests per vehicle per month, we’re already starting to flex the basic $10. Of course, you also want to track your trip data and output that - so add in another 500-ish requests per vehicle based on the amount of driving and interval levels.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">We’re looking at $8 in wake-ups, $4 for commands, $8 for data, and at least $1 in Streaming Signals. Just for two vehicles with a fairly basic set of data tracking and usage of commands, we’re already past the $10 mark.</p><div style="box-sizing: border-box; margin-left: 81.2188px; margin-right: 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></div><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">If you’re keeping track, that’s a total of $21 per month—minus the $10 discount Tesla provides per account. At its current usage, our Home Assistant integration would require us to pay $11 for API access a month—as much as Premium Connectivity.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">While switching to the streaming API is the answer here, it’ll take time for tools and services to transition to the new way of gathering data. Unfortunately, Tesla isn’t giving developers and home users a lot of time, with pricing taking effect on January 1st, 2025.</p><h2 style="box-sizing: border-box; line-height: 1.2; font-size: 28px; margin: 20px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Third-Party Developers</h2><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Take all those fairly conservative numbers I’ve provided - and scale those up 3-4x. That’s how much data is requested by popular third-party apps. They’re looking at monthly costs well beyond $50 per vehicle, with current data draws. Again, they’ll need to transition to streaming APIs where possible and optimize the data they gather, but Tesla isn’t giving developers a lot of time to do so.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Third-party developers will need to find ways to optimize their workflows, potentially raise prices, and work within and around Tesla’s new strict rules on pulling data from vehicles.</p><h2 style="box-sizing: border-box; line-height: 1.2; font-size: 28px; margin: 20px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Vehicle Support</h2><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Unfortunately, 2020 and older Model S and Model X vehicles don’t support the streaming API, which means that they may not be supported by most third-party services going forward.</p><p style="box-sizing: border-box; line-height: 34px; font-size: 18px; margin: 18px 81.2188px; color: rgb(65, 65, 65); font-family: &quot;Open Sans&quot;, -apple-system, &quot;system-ui&quot;, Roboto, Arial, &quot;Helvetica Neue&quot;, Helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">We’re hoping Tesla takes some steps to make its API more flexible and affordable, as even simple home use can get quite expensive.</p>After several years of allowing third-party developers and tinkerers free access to the Tesla API, Tesla has finally announced a pricing for their API. Back in 2023, Tesla introduced their [new API system](https://www.notateslaapp.com/news/1653/tesla-creates-official-apis-for-third-party-services-to-start-charging-for-usage), along with them being “temporarily free.” 

Back then, Tesla announced their Discovery tier, which had some limitations but set the groundwork for what was to come. For the first time, developers had official documentation and support, but pricing still was to be determined.

Tesla recently announced pricing for its APIs, which, unfortunately, is much higher than many were expecting. We’ll start by reviewing everything that’s been announced and then explore what could happen for both home users and larger third-party apps.

Fleet API - Usage Based Pricing
Tesla is implementing a pay-per-use pricing model that charges users based on overall usage. Some items, such as Streaming Signals and Commands, are at a lower cost, which helps to incentivize developers to be smart with the data they’re pulling from vehicles. Meanwhile, vehicle data via REST APIs and waking up vehicles is much more expensive.

Here’s the pricing chart:

Data Type

Signals/Requests

Cost

Streaming Signals

150,000

$1

Commands

1,000

$1

Data

500

$1

Wakes

50

$1

Tesla seems like they want developers to focus on what’s happening in the now - rather than being able to track a long driving session or pull charging history from the vehicle. While some things seem cheap and some seem expensive, the overall cost is still fairly high, even for limited personal use.

Additionally, the new Fleet Telemetry doesn’t offer all the data points that are available from the older vehicle data API. That means that you won’t always be able to take advantage of the cheaper “Streaming Signals” data type - but have to fall back to the data package - which is far more expensive.

Many services use REST APIs for vehicle data at about a one-minute interval, which means that each hour of driving or charging could cost about $0.12. Since the average vehicle drives for about an hour a day and charges for an hour or two, that could cost about $0.36 cents per day — per vehicle.

Looking at a month’s worth, that’s about $10/month per vehicle. While the streaming signal API is much cheaper, the total cost would vary depending on the interval and data being collected. Each signal is a piece of data, so if you’re tracking the vehicle’s speed, location, odometer, state and battery level, that’s five signals.

Fleet API - Personal Use
Tesla’s Fleet API provides each Tesla account with a $10 monthly discount - which Tesla says can cover data streaming, 100 commands, and two wakes per day for two vehicles per month. There isn’t a discount per vehicle - it's per account, so if you have more vehicles, it’s less beneficial to you. This should allow home tinkerers to create some automations and track some data without having to pay for access.

However, this is a fairly restrictive amount of signals/requests for personal use - let’s look at the author’s Home Assistant integration for a fairly simple example.

On average, with a Home Assistant tablet at home to start/stop charging, check charge states, precondition vehicles, send Google Map targets, and a few more simple commands for just two vehicles - we make, on average, about 200 requests a week per vehicle. That comes out to 1,600 requests a month and 400 wake-ups a month. 

We’re also requesting data from the Data portion of the API - not the Streaming Signals portion of the API - and that’s about every five minutes while charging - so a wake isn’t needed. With 480 data requests per vehicle per month, we’re already starting to flex the basic $10. Of course, you also want to track your trip data and output that - so add in another 500-ish requests per vehicle based on the amount of driving and interval levels.

We’re looking at $8 in wake-ups, $4 for commands, $8 for data, and at least $1 in Streaming Signals. Just for two vehicles with a fairly basic set of data tracking and usage of commands, we’re already past the $10 mark.

If you’re keeping track, that’s a total of $21 per month—minus the $10 discount Tesla provides per account. At its current usage, our Home Assistant integration would require us to pay $11 for API access a month—as much as Premium Connectivity.

While switching to the streaming API is the answer here, it’ll take time for tools and services to transition to the new way of gathering data. Unfortunately, Tesla isn’t giving developers and home users a lot of time, with pricing taking effect on January 1st, 2025.

Third-Party Developers
Take all those fairly conservative numbers I’ve provided - and scale those up 3-4x. That’s how much data is requested by popular third-party apps. They’re looking at monthly costs well beyond $50 per vehicle, with current data draws. Again, they’ll need to transition to streaming APIs where possible and optimize the data they gather, but Tesla isn’t giving developers a lot of time to do so.

Third-party developers will need to find ways to optimize their workflows, potentially raise prices, and work within and around Tesla’s new strict rules on pulling data from vehicles.

Vehicle Support
Unfortunately, 2020 and older Model S and Model X vehicles don’t support the streaming API, which means that they may not be supported by most third-party services going forward.

We’re hoping Tesla takes some steps to make its API more flexible and affordable, as even simple home use can get quite expensive.
Source : https://www.notateslaapp.com/news/2415/tesla-announces-api-pricing-third-party-service-costs-expected-to-rise

**Comment by cwanja** [comment]
See #4408
